### DISCUSSION TITLE
Add ClickHouse columnar database as data-source

### DISCUSSION THREAD

**Discussion by Slach** [chartbrew/chartbrew#159] [Ideas]
Thanks a lot for you great efforts and nice product
please look to http://clickhouse.yandex/docs/en/
ClickHouse is a blazing fast open source Columnar database for Analytics

could you add ClickHouse as data source in your product?

**Comment by razvanilin** [comment]
Hey! Thanks for dropping by and suggesting the new data source. I wasn't aware of ClickHouse before, but it's definitely doable to add it as another option.

In the meantime, if you have an API that can query the data from your database, you could use the API connection to fetch the data.

**Comment by Slach** [comment]
any progress with this issue?

**Comment by razvanilin** [comment]
hey @Slach I don't have the bandwidth to look into this at the moment as it seems to be quite a big undertaking.

Although, I went through the website a bit and noticed that it has a MySQL interface. Would it be possible to use the MySQL connection from Chartbrew to connect to your ClickHouse data source? https://clickhouse.tech/docs/en/interfaces/mysql/

**Reply by Niek** [reply]
I tried your suggestion, after [enabling the MySQL protocol adapter in ClickHouse](https://clickhouse.com/docs/en/interfaces/mysql#enabling-the-mysql-interface-on-self-managed-clickhouse) I can connect locally like this:
```bash
mysql -h 127.0.0.1 -u default -P 9004 default
```
Unfortunately I can't connect from Chartbrew, I'm using the connecting string:
```url
mysql://default:@127.0.0.1:9004/default
```
After clicking `Test connection` I get:
```
Code: 115. DB::Exception: Setting time_zone is neither a builtin setting nor started with the prefix 'SQL_' registered for user-defined settings. (UNKNOWN_SETTING) (version 25.1.3.23 (official build))
```
Are there any plans to skip/disable the timezone settings per connection?

**Reply by Niek** [reply]
Alternatively, maybe a `ClickHouse MySQL` `subType` can be added, similar to the `RDS MySQL` that you have?

**Reply by razvanilin** [reply]
Hey @Niek I was just looking into this now and it looks like ClickHouse doesn't support all default MySQL settings and syntax used by Sequelize (the current ORM used under the hood by Chartbrew)

I managed to get past the error with the time_zone, but keep running into other issues down the line. I think it's best to create a separate `clickhouse` type in Chartbrew that's using the native clickhouse driver instead.

I added this in the roadmap to keep track: https://chartbrew.productroad.com/board/feature-requests/add-clickhouse-connection/

**Reply by Niek** [reply]
Thanks for lookingji into this @razvanilin! Official ClickHouse support would be a great feature to have. One thing I forgot to mention is that you get Cloudflare Analytics Engine support for free as well, since it uses ClickHouse under the hood (over the HTTP protocol though).

**Reply by razvanilin** [reply]
Ah nice, I didn't actually have CF's analytics engine on my radar before so that's good to know!

**Comment by razvanilin** [comment]
ClickHouse now available in v3.13.0 🎉

https://github.com/chartbrew/chartbrew/releases/tag/v3.13.0

**Comment by Slach** [comment]
great news, thanks for your efforts!
